### PR TITLE
[commissioner] add 'state' CLI command

### PR DESCRIPTION
**Pull request by jwhui**

### CODE REVIEW

File: src/posix/platform/netif.cpp [hunk 1]
```diff
@@ -292,8 +292,8 @@ static void InitNetaskWithPrefixLength(struct in6_addr *address, uint8_t prefixL
 
 static uint8_t NetmaskToPrefixLength(const struct sockaddr_in6 *netmask)
 {
-    return otIp6PrefixMatch(static_cast<const otIp6Address *>(netmask->sin6_addr.s6_addr),
-                            static_cast<const otIp6Address *>(allOnes));
+    return otIp6PrefixMatch(reinterpret_cast<const otIp6Address *>(netmask->sin6_addr.s6_addr),
+                            reinterpret_cast<const otIp6Address *>(allOnes));
```
Comment (abtink) [inline src/posix/platform/netif.cpp:296]: 👍

File: src/cli/cli_commissioner.cpp [hunk 2]
```diff
@@ -345,7 +345,11 @@ void Commissioner::HandleStateChanged(otCommissionerState aState, void *aContext
 void Commissioner::HandleStateChanged(otCommissionerState aState)
 {
     mInterpreter.mServer->OutputFormat("Commissioner: ");
+    PrintState(aState);
```
Comment (simonlingoogle) [inline src/cli/cli_commissioner.cpp:348]: I think `PrintState` should not print "\r\n".
```suggestion
    PrintState(aState);
    mInterpreter.mServer->OutputFormat("\r\n");
```

File: src/cli/cli_commissioner.cpp [hunk 3]
```diff
@@ -411,6 +415,16 @@ otError Commissioner::ProcessStop(uint8_t aArgsLength, char *aArgs[])
     return otCommissionerStop(mInterpreter.mInstance);
 }
 
+otError Commissioner::ProcessState(uint8_t aArgsLength, char *aArgs[])
+{
+    OT_UNUSED_VARIABLE(aArgsLength);
+    OT_UNUSED_VARIABLE(aArgs);
+
+    PrintState(otCommissionerGetState(mInterpreter.mInstance));
```
Comment (simonlingoogle) [inline src/cli/cli_commissioner.cpp:423]:
```suggestion
    PrintState(otCommissionerGetState(mInterpreter.mInstance));
    mInterpreter.mServer->OutputFormat("\r\n");
```

File: src/posix/platform/netif.cpp (hunk 1 above)
Comment (jwhui) [reply src/posix/platform/netif.cpp]: Oops. I meant to push this in a separate PR :)

File: src/cli/cli_commissioner.cpp [hunk 4]
```diff
@@ -345,17 +345,22 @@ void Commissioner::HandleStateChanged(otCommissionerState aState, void *aContext
 void Commissioner::HandleStateChanged(otCommissionerState aState)
 {
     mInterpreter.mServer->OutputFormat("Commissioner: ");
+    PrintState(aState);
+    mInterpreter.mServer->OutputFormat("\r\n");
+}
 
+void Commissioner::PrintState(otCommissionerState aState)
```
Comment (wgtdkp) [inline src/cli/cli_commissioner.cpp:352]: nit: Looks like having a function coverts the `State` to its string representation can make the printing code simpler :)

Comment (jwhui) [reply src/cli/cli_commissioner.cpp:352]: Are you proposing to make this a static function rather than a private method?

Comment (wgtdkp) [reply src/cli/cli_commissioner.cpp:352]: oh, maybe a function like this:
```C++
const char *Commissioner::StateToString(otCommissionerState aState)
{
    switch (aStahe)
    {
    case kDisabled: return "Disabled";
    ...
    }
}
```

and we can simplify the printing code:
```c++
    mInterpreter.mServer->OutputFormat("Commissioner: %s\r\n", StateToString(aState));
```

**Pull request merged**